source/codegen/service_helpers.py

```diff
@@ -731,3 +731,13 @@ def get_c_api_name(function_name) -> str:
         base_name = function_name[len("Begin") :]
         return f"{base_name}"
     return f"{function_name}"
+
+
+def get_output_streaming_params_to_define(parameters, streaming_param) -> List[dict]:
+    """Get the list of output streaming parameter that needs to be defined."""
+    params = []
+    output_params = [p for p in parameters if common_helpers.is_output_parameter(p)]
+    for param in output_params:
```

Comment:
Nit: the list comprehension for selecting `output_params` is much cleaner than the loop for selecting streaming params. Using list comprehensions and helper methods makes a big difference:

```python

def is_streaming_param_except_name(param, name_to_avoid_for_some_reason):
   # TODO why avoid name_to_avoid_for_some_reason?
   param.get("is_streaming_type", False) and param["name"] != name_to_avoid_for_some_reason

def get_output_streaming_params_to_define(parameters, streaming_param) -> List[dict]:
    """Get the list of output streaming parameter that needs to be defined."""
    output_params = [p for p in parameters if common_helpers.is_output_parameter(p)]
    return [p for p in output_params if is_streaming_param(p, streaming_param["name"])]
```

Actually, rewriting this, I realize I don't understand this code. We're avoiding one specific streaming parameter? Why is it not sufficient to look for `is_output_parameter` and `is_streaming_type`? Is `streaming_param` named correctly?